WordPress/gutenberg
# Site Editor: Redirect `/wp_template_part/all` to `/patterns`

Follow-up #61080 
See this comment: https://github.com/WordPress/gutenberg/pull/61080#issuecomment-2085893842

## What?

This PR ensures that when `/wp_template_part/all` path is accessed, it will redirect to `/patterns`.

## Why?

The All Template Parts page no longer exist, as template parts have already been integrated into the Patterns page. However, for backward compatibility, the Patterns page was displayed when this path was accessed. We should be able to completely remove references to `/wp_template_part/all` by redirecting to the `/patterns` path.

## How?

Added redirect to `RouterProvider`. It works fine as far as I've tested it, but I'd appreciate some advice if this is the ideal approach.

## Testing Instructions

- Enter `http://localhost:8888/wp-admin/site-editor.php?path=/wp_template_part/all` in the URL of your browser.
- You should be redirected to `http://localhost:8888/wp-admin/site-editor.php?path=/patterns`.
- You should now see a sidebar and data view for the Patterns page.
